frame/lending/src/mocks_offchain/mod.rs

```diff
@@ -0,0 +1,512 @@
+	pub ExistentialDeposits: |_currency_id: CurrencyId| -> Balance {
+		Zero::zero()
+	};
+}
+
+parameter_types! {
+	pub MaxLocks: u32 = 2;
+}
+
+impl orml_tokens::Config for Runtime {
+	type Event = Event;
+	type Balance = Balance;
+	type Amount = Amount;
+	type CurrencyId = CurrencyId;
+	type WeightInfo = ();
+	type ExistentialDeposits = ExistentialDeposits;
+	type OnDust = ();
+	type MaxLocks = ();
+	type DustRemovalWhitelist = Everything;
+}
+
+ord_parameter_types! {
+	pub const RootAccount: AccountId = *ALICE;
+}
+
+pub struct NoopRegistry;
+
+impl<CurrencyId, AccountId> GovernanceRegistry<CurrencyId, AccountId> for NoopRegistry {
+	fn set(_k: CurrencyId, _value: SignedRawOrigin<AccountId>) {}
+}
+
+impl<CurrencyId> GetByKey<CurrencyId, Result<SignedRawOrigin<AccountId>, sp_runtime::DispatchError>>
+	for NoopRegistry
+{
+	fn get(_k: &CurrencyId) -> Result<SignedRawOrigin<AccountId>, sp_runtime::DispatchError> {
+		Ok(SignedRawOrigin::Root)
+	}
+}
+
+impl pallet_assets::Config for Runtime {
+	type NativeAssetId = NativeAssetId;
+	type GenerateCurrencyId = LpTokenFactory;
+	type AssetId = CurrencyId;
+	type Balance = Balance;
+	type NativeCurrency = Balances;
+	type MultiCurrency = Tokens;
+	type WeightInfo = ();
+	type AdminOrigin = EnsureSignedBy<RootAccount, AccountId>;
+	type GovernanceRegistry = NoopRegistry;
+}
+
+parameter_types! {
+	pub const MinBalance: Balance = 0;
+	pub const MinU32: u32 = 0;
+	pub const MinU64: u64 = 0;
+}
+
+pub struct Decimals;
+impl LocalAssets<CurrencyId> for Decimals {
+	fn decimals(_currency_id: CurrencyId) -> Result<Exponent, DispatchError> {
+		Ok(12)
+	}
+}
+
+impl pallet_oracle::Config for Runtime {
+	type Event = Event;
+	type Currency = Assets;
+	type AssetId = CurrencyId;
+	type PriceValue = Balance;
+	type AuthorityId = AuthorityId;
+	type MinStake = MinBalance;
+	type StakeLock = MinU64;
+	type StalePrice = MinU64;
+	type AddOracle = EnsureSignedBy<RootAccount, AccountId>;
+	type MaxAnswerBound = MinU32;
+	type MaxAssetsCount = MinU32;
+	type MaxHistory = MinU32;
+	type MaxPrePrices = MinU32;
+	type WeightInfo = ();
+	type LocalAssets = Decimals;
+
+	type TreasuryAccount = RootAccount;
+}
+
+impl DeFiComposableConfig for Runtime {
+	type MayBeAssetId = CurrencyId;
+	type Balance = Balance;
+}
+
+parameter_types! {
+	pub DutchAuctionPalletId: PalletId = PalletId(*b"dutchauc");
+}
+
+// later will reuse mocks from that crate
+pub struct DutchAuctionsMocks;
+
+impl WeightToFeePolynomial for DutchAuctionsMocks {
+	type Balance = Balance;
+
+	fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
+		let one = WeightToFeeCoefficient {
+			degree: 1,
+			coeff_frac: Perbill::zero(),
+			coeff_integer: WEIGHT_TO_FEE.with(|v| *v.borrow()),
+			negative: false,
+		};
+		smallvec![one]
+	}
+}
+
+pub struct XcmFake;
+impl Into<Result<cumulus_pallet_xcm::Origin, XcmFake>> for XcmFake {
+	fn into(self) -> Result<cumulus_pallet_xcm::Origin, XcmFake> {
+		todo!("please test via local-integration-tests")
```

Comment:
if these methods will never be implemented, `unimplemented!` would be more semantic here

Reply:
Done for tests.rs and tests_offchain.rs